.github/workflows/fail_on_error.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+import json
+import sys
+
+# Return whether SARIF file contains error-level results
+def codeql_sarif_contain_error(filename):
```

Comment:
Most Python code in the repository is old, but since we are on Python 3 now, please add type hints to the function. (I'm guessing `..(filename: str) -> bool:`).